### ISSUE TITLE
Incorrect timezone in Util::getMicrotimeFloatDateTime

### ISSUE THREAD

**Issue by fliespl** [mmucklo/DtcQueueBundle#126]
When using createFromFormat with microtime there is an issue with it that it ignores timezone in such case.

Correct behaviour is to:
- create DateTime object without setting timezone
- setting timezone using setters

Current DTC way:
```
date_default_timezone_set('Europe/Berlin');
$tz = new \DateTimeZone(date_default_timezone_get());
$date = \DateTime::createFromFormat('U.u', number_format($microtime, 6, '.', ''), $tz);
var_dump($date);
```

Result:
```
class DateTime#2165 (3) {
  public $date =>
  string(26) "2020-03-18 08:02:05.241020"
  public $timezone_type =>
  int(1)
  public $timezone =>
  string(6) "+00:00"
}
```

Setting timezone using setter.
```
date_default_timezone_set('Europe/Berlin');
$date = \DateTime::createFromFormat('U.u', number_format($microtime, 6, '.', ''));
$date->setTimezone($tz);
var_dump($date);
```
Result:
```
class DateTime#2166 (3) {
  public $date =>
  string(26) "2020-03-18 09:02:05.241020"
  public $timezone_type =>
  int(3)
  public $timezone =>
  string(13) "Europe/Berlin"
}
```

I can create pull request if requested.

**Comment by mmucklo** [comment]
Wow,

I just reproduced that as well:

```php
<?php

date_default_timezone_set('Europe/Berlin');
$tz = new \DateTimeZone(date_default_timezone_get());
var_dump($tz);
$date = \DateTime::createFromFormat('U', '123456789', $tz);
var_dump($date);
```

```
object(DateTimeZone)#1 (2) {
  ["timezone_type"]=>
  int(3)
  ["timezone"]=>
  string(13) "Europe/Berlin"
}
object(DateTime)#2 (3) {
  ["date"]=>
  string(26) "1973-11-29 21:33:09.000000"
  ["timezone_type"]=>
  int(1)
  ["timezone"]=>
  string(6) "+00:00"
}
```

It's odd because the spec for createFromFormat takes a DateTimeZone object as the third parameter:

https://www.php.net/manual/en/datetime.createfromformat.php

**Comment by axxox** [comment]
> It's odd because the spec for createFromFormat takes a DateTimeZone object as the third parameter:
> 
> https://www.php.net/manual/en/datetime.createfromformat.php

@mmucklo Check the note in `timezone` parameter description: Note:

> The timezone parameter and the current timezone are ignored when the time parameter either contains a UNIX timestamp (e.g. 946684800) or specifies a timezone (e.g. 2010-01-28T15:00:00+02:00).

That's why current solution doesn't work.

**Comment by fliespl** [comment]
Looks great - thanks!

**Comment by mmucklo** [comment]
Should be fully fixed as of version 6.0.6.
